src/boot/efi/log.c

```diff
@@ -58,6 +59,31 @@ void efi_assert(const char *expr, const char *file, unsigned line, const char *f
         freeze();
 }
 
+#if defined(__i386__) || defined(__x86_64__)
+enum {
+        DEBUGCON_PORT = 0x402,
+        DEBUGCON_MAGIC = 0xe9,
+};
```

Comment:
i am not a fan of (mis-)using anonymous enums for this. enums are by default signed, but inb() both takes and returns an unsigned. So you always have this weird cast there. in particular as inb returns you an 8bit value (unsigned), but DEBUGCON_MAGIC is signed, but outside of the signed 8bit value.

just use a macro, and suffix this with U (or enclose in `UINT8_C()`), like everybody else, then the types are clear